crd/overlayextensionconfig/api/v1alpha1/overlayextensionconfig_types.go

```diff
@@ -0,0 +1,59 @@
+//go:build !ignore_uncovered
+// +build !ignore_uncovered
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OverlayExtensionConfig is the Schema for the overlayextensionconfigs API
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:shortName=oec
+// +kubebuilder:printcolumn:name="OverlayExtensionConfig IP range",type=string,priority=1,JSONPath=`.spec.extensionIPRange`
+type OverlayExtensionConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OverlayExtensionConfigSpec   `json:"spec,omitempty"`
+	Status OverlayExtensionConfigStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// OverlayExtensionConfigList contains a list of OverlayExtensionConfig
+type OverlayExtensionConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []OverlayExtensionConfig `json:"items"`
+}
+
+// OverlayExtensionConfigSpec defines the desired state of OverlayExtensionConfig.
+type OverlayExtensionConfigSpec struct {
+	// ExtensionIPRange field defines a CIDR that should be able to reach routing domain ip addresses.
+	// +kubebuilder:validation:Optional
+	ExtensionIPRange string `json:"extensionIPRange,omitempty"`
```

Comment:
Should we add the NetworkGUID here and make it optional.

Reply:
We can get vnetGUID from cluster node labels in OEC controller. Why would we want it in the CRD? Or do you think we might need it in the future to support vnet peering?

Reply:
@paulgmiller , what are your thoughts on this? Should we include networkGUID now or bump the crd version later if we need it for vnet peering?

Reply:
Decided to not include networkGUID field for now.